core/src/test/java/dev/morphia/test/mapping/codec/pojo/PropertyCodecProviderTest.java

```diff
@@ -0,0 +1,113 @@
+package dev.morphia.test.mapping.codec.pojo;
```

Comment:
In general, this test does more than it needs to. It would be much simpler to have your custom provider throw an exception which can be caught to validate that it is properly registered with the `CodecProvider`.  This removes the need to widen the visibility of the internal types as you've done here.

Reply:
done